### PR TITLE
Add https to local

### PR DESCRIPTION
**Pull request by rpsu**
Add SSL/TLS cert creation to init step. Adjust docker-compose file templates and Traefik configuration to handle certs and HTTPS connections. Terminate all connections in Treafik and pass on decrypted requests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,6 +110,19 @@ subdomains:
 - Whoami (**only** request headers ) -
   [http://**whoami**.example.com](http://whoami.example.com)
 
+#### SSL/TLS certificates, ie. HTTPS for local
+
+During the initial project initialisation local-docker generates self-signed SSL/TLS cerificates for your local development needs. Certificate is created for `*.example.local` -domain, which means it covers all auto-generated subdomains you have.
```
Comment (kekkis) [inline README.md]: Is this literally speaking true, meaning does it generate for `*.example.local` literally? It does not generate for the domain name(s) set in the config?

File: README.md [hunk 2]
```diff
@@ -110,6 +110,19 @@ subdomains:
...
+
+If you wish to have automatic HTTP -> HTTPS redirection for you local check your `docker-compose.yml` file and comment out some lines below `label` -key (under `nginx` or any other service).
```
Comment (kekkis) [inline README.md]:
```suggestion
If you wish to have automatic HTTP -> HTTPS redirection for you local check your `docker-compose.yml` file and comment out some lines below `label` key (under `nginx` or any other service).
```

Unify the absence of hyphens

File: README.md [hunk 3]
```diff
@@ -164,8 +177,7 @@ done via your site's `settings.php` file. Where Skeleton may use
 and by default credentials `drupal`/`drupal` with database named
 `drupal` (simple!). Example for Drupal 8:
 
-    <?php
-      $databases['default']['default'] = [
+    $databases['default']['default'] = [
```
Comment (kekkis) [inline README.md]: Seems like an unnecessary change? But if on purpose, maybe outdent the following lines too, to keep the indentation at 2 spaces?

File: README.md [hunk 4]
```diff
@@ -180,6 +192,47 @@ However, editing `settings.php` manually for database connection is
...
+Place this inside your `sites/XXX/settings.php` file to fix the issue:
+
+    /**
+     * Tell all Drupal sites that we're running behind an HTTPS proxy.
+     *
+     * @see https://www.drupal.org/node/425990
+     */
+
+    // Drupal 7 configuration.
+    if (explode('.', VERSION)[0] == 7) {
```
Comment (kekkis) [inline README.md]: This won't run cleanly on Drupal 8+, there is no defined VERSION. Improvement suggestion, depending on usecase:
```suggestion
    if (defined('VERSION') && explode('.', VERSION)[0] == 7) {
```

File: docker/scripts/ld.command.tls-cert.sh [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    echo " "
+    echo "You can re-generate the files any time. Should you need more than one certificate"
+    echo "you can rename existing files and generate new ones using command "
+    echo "$ ./ld tls-cert"
+    echo " "
+    echo "Remember to update also file ./docker/certs/certs.yml to contain all the certificates"
+    echo "your project needs."
+    echo " "
+    echo "NOTE: By default Traefik configuration is set to handle also HTTPS requests, but "
+    echo "rediretion HTTP -> HTTPS (nginx, mailhog, solr, etc.) need docker-compose.yml file "
```
Comment (kekkis) [inline docker/scripts/ld.command.tls-cert.sh]:
```suggestion
    echo "redirection HTTP -> HTTPS (nginx, mailhog, solr, etc.) need docker-compose.yml file "
```

File: docker/scripts/ld.command.tls-cert.sh [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+function ld_command_tls-cert_extended_help() {
+    echo "Create local TLS certificate (HTTPS)"
+    echo " "
+    echo "Certificates and the configuration are created during init -phase"
+    echo "prefixed with the project neme and stored in docker/certs -folder."
+    echo "Traefik learns about your HTTPS certificates via ./docker/certs/certs.yml -file"
+    echo "(mounted volume)."
+    echo " "
+    echo "Certificate is a multidomain certificate: "
+    echo "    *.example.com"
```
Comment (kekkis) [inline docker/scripts/ld.command.tls-cert.sh]: Could this use `$LOCAL_DOMAIN` instead of `example.com`?

File: README.md (hunk 1 above)
Comment (kekkis) [reply README.md]: Judging by the script files, no. It will be created for the configured domain.

Comment (rpsu) [reply README.md]: yes, surely not literally `*.example.com` but similar cathcall cert for the local domain. Do you think this needs to be updated, to be more explicit? `README.md` uses `example.com` all over in similar contexts.

File: README.md (hunk 3 above)
Comment (rpsu) [reply README.md]: this change is mainly to not confuse some IDE's (PHPStorm I'm looking at you) when they render Markdown. But the indentation did not follow up DCS, so that is now updated, too, with 5254943

File: README.md (hunk 4 above)
Comment (rpsu) [reply README.md]: Protective coding yes :) 

But mind you, the `VERSION` constant is always present in *Drupal*.

File: docker/scripts/ld.command.tls-cert.sh (hunk 6 above)
Comment (rpsu) [reply docker/scripts/ld.command.tls-cert.sh]: yes, good idea. 

Extended help hooks may be called even when the project is not yet initialized, so I'll add also conditional config reading if the project has been initialized.

Comment (rpsu) [reply docker/scripts/ld.command.tls-cert.sh]: 7461d1f and 17662c3

File: README.md (hunk 1 above)
Comment (kekkis) [reply README.md]: It's probably ok to leave as is. I was reading this only in the context of these changes. Thanks for the confirmation regarding README.
